splink/postgres/postgres_linker.py

```diff
@@ -0,0 +1,246 @@
+        if isinstance(input, str):
+            return self._table_to_splink_dataframe(table_name, input)
+
+        # Check if table name is already in use
+        exists = self._table_exists_in_database(table_name)
+        if exists:
+            if not overwrite:
+                raise ValueError(
+                    f"Table '{table_name}' already exists in database. "
+                    "Please use the 'overwrite' argument if you wish to overwrite"
+                )
+            else:
+                self._delete_table_from_database(table_name)
+
+        if isinstance(input, dict):
+            input = pd.DataFrame(input)
+        elif isinstance(input, list):
+            input = pd.DataFrame.from_records(input)
+
+        def connection_fairy():
+            return self.con
+
+        engine = create_engine("postgresql://", creator=connection_fairy)
+
+        # Will error if an invalid data type is passed
+        input.to_sql(
+            table_name,
+            con=engine,
+            index=False,
+            if_exists="replace",
+            schema=self._db_schema,
+        )
+        return self._table_to_splink_dataframe(table_name, table_name)
+
+    def _random_sample_sql(self, proportion, sample_size, seed=None):
+        if proportion == 1.0:
+            return ""
+        if seed is None:
+            seed = "random()"
+
+        sample_size = int(sample_size)
+        return f"""
+                WHERE unique_id IN (
+                    SELECT unique_id
+                    FROM __splink__df_concat_with_tf"
+                    ORDER BY {seed}
+                    LIMIT {sample_size}
+                )
+            """
+
+    @property
+    def _infinity_expression(self):
+        return "'infinity'"
+
+    def _table_exists_in_database(self, table_name):
+        sql = f"""
+        SELECT table_name
+        FROM information_schema.tables
+        WHERE table_schema = '{self._db_schema}'
+        AND table_name = '{table_name}';
+        """
+
+        cur = self.con.cursor(cursor_factory=RealDictCursor)
+        cur.execute(sql)
+        rec = cur.fetchone()
+        cur.close()
+        return rec is not None
+
+    def _delete_table_from_database(self, name):
+        drop_sql = f"DROP TABLE IF EXISTS {name};"
+        cur = self.con.cursor()
+        cur.execute(drop_sql)
+        self.con.commit()
+        cur.close()
+
+    def _create_log2_function(self):
+        sql = """
+        CREATE OR REPLACE FUNCTION log2(n float8)
+        RETURNS float8 AS $$
+        SELECT log(2.0, n::numeric)::float8;
+        $$ LANGUAGE SQL IMMUTABLE;
+        """
+        cur = self.con.cursor()
+        cur.execute(sql)
+        self.con.commit()
+        cur.close()
+
+    def _create_splink_schema(self):
+        sql = f"""
+        CREATE SCHEMA IF NOT EXISTS {self._db_schema};
+        """
+        cur = self.con.cursor()
+        cur.execute(sql)
+        self.con.commit()
+        cur.close()
```

Comment:
I wonder if these could make use of `_run_sql_execution` instead of manually using the cursor each time?